build/index.ts

```diff
@@ -359,6 +360,19 @@ export async function buildDocument(
     browserCompat &&
     (Array.isArray(browserCompat) ? browserCompat : [browserCompat]);
 
+  if (doc.browserCompat) {
+    for (const feature of Object.values<WebFeature>(webFeatures)) {
+      if (
+        feature.status &&
+        feature.compat_features?.some((query) =>
+          doc.browserCompat?.includes(query)
+        )
+      ) {
+        doc.baseline = feature.status;
+      }
+    }
+  }
```

Comment:
🟡 Can we abstract this away into a processor function `deriveBaseline(doc)` or similar?